src/library_wasmfs.js

```diff
@@ -0,0 +1,17 @@
+var WasmfsLibrary = {
+  $wasmfsBuffers: [null, [], []],
+  emscripten_wasmfs_printbuffer__deps: ['$wasmfsBuffers'],
+  emscripten_wasmfs_printbuffer: function(stream, ptr, len) {
+    for (var j = 0; j < len; j++) {
+      if (HEAPU8[ptr+j] === 0 || HEAPU8[ptr+j] === 10) {
+        (stream === 1 ? out : err)(UTF8ArrayToString(wasmfsBuffers[stream], 0));
+        wasmfsBuffers[stream].length = 0;
+      } else {
+        wasmfsBuffers[stream].push(HEAPU8[ptr+j]);
+      }
+    }
+    
+  }
+}
```

Comment:
I would recommend implementing this whole function in C/C++. That way the state will be multithreading safe from the get-go (e.g. for behavior like `fflush()`ing stdout/stderr from another thread than the one(s) that have done partial printing). That method could use e.g. `emscripten_console_log()` and `emscripten_console_error()` for the printing.